### ISSUE TITLE
Extensions: Upgrading Extensions Blows Up

### ISSUE THREAD

**Issue by jpoon** [Microsoft/vscode#1111]
I pushed out a new version of the Vim extension, however, when upgrading it on top of the existing version, we are seeing this error: https://github.com/VSCodeVim/Vim/issues/93. Installing the new version by itself (with no previous versions of the Vim extension) works successfully. 

I've been unsuccessful at getting a repro. Is there a way in which I can install a specific version from the marketplace?

I've built the packages (`vsce package`) and made them available here: https://github.com/VSCodeVim/Vim/releases. I have tried:

1) `code vim-0.0.2.vsix` - This installs successfully
2) On top of the existing install `code vim-0.0.3.vsix`, VSCode blows up with a different error than reported in https://github.com/VSCodeVim/Vim/issues/93. The [console log is here](https://github.com/Microsoft/vscode/files/55978/vscode-blows-up.txt).
3) Installing `code vim-0.0.3.vsix` (not upgrading) works fine

tldr;
* Is there a way to install a specific version of an extension from the marketplace?
* Barring the above, is manually upgrading a plugin through the vsix expected to work?

**Comment by jpoon** [comment]
Why hello @joaomoreno. In investigating this, I'd like to point out that user error is plausible :)

**Comment by joaomoreno** [comment]
Hey hey, thanks for the issue! There currently isn't a way to install a specific version, unfortunately. As for the second question, it should work! Will investigate...

**Comment by jpoon** [comment]
Got another report of the 2nd question (manually upgrading a plugin): https://github.com/VSCodeVim/Vim/issues/93#issuecomment-163345243

**Comment by remcoros** [comment]
I ran into this developing my extension as well (vscode-startanyshell).

Manually updating (vsix) or trough command palette, leaves the extension folder in a 'broken' state. The extension folder does not contain all files. Like if uncompressing the package stopped half-way.

I really tought it was just my vscode installation or machine, since I didn't see any issues about this.

Look like I'm not the only one though.

I didn't investigate further. Anything I can do to help? 

b.t.w. This happens on my machine with ALL extensions I try to update.
My current solution is to just delete the whole extension folder I want to update, then install it.

**Comment by joaomoreno** [comment]
@jpoon @remcoros which OS are you on?

**Comment by jpoon** [comment]
Windows 10

**Comment by remcoros** [comment]
Windows 10 also

**Comment by pagebrooks** [comment]
I'm seeing the same issue as @remcoros 
OS: Windows 7 x64
Extension: https://github.com/pagebrooks/vscode-hl7

**Comment by joaomoreno** [comment]
Alright, super reproducible. Let's see if we can get a nice fix for December.

**Comment by jpoon** [comment]
@joaomoreno Thanks for looking into it.
